### PR TITLE
Fix crashes when instrumenting malloc/free.

### PR DESCRIPTION
**Pull request by danielfenner**
Bug: http://b/207361417
Test: A lot of manual testing. Will add these to e2e tests.

### CODE REVIEW

File: src/UserSpaceInstrumentation/InstrumentProcess.cpp [hunk 1]
```diff
@@ -66,6 +67,17 @@ bool ProcessWithPidExists(pid_t pid) {
   return result.value();
 }
 
+// We need to initialize some thread local memory when entering the payload functions. This leads to
+// a situation where instrumenting the functions below would lead to a recursive call into the
+// instrumentation. We just silently skip these and leave instrumenting them to the kernel/uprobe
+// fallback.
+bool IsBlacklisted(std::string_view function_name) {
```
Comment (beckerhe) [inline src/UserSpaceInstrumentation/InstrumentProcess.cpp:74]: I think we should go with inclusive language here and call it `IsBlocklisted` or something along those lines.

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/UserSpaceInstrumentation/InstrumentProcess.cpp (hunk 1 above)
Comment (danielfenner) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp]: done

Comment (beckerhe) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp]: thx

**Review by dpallotti — APPROVED** [review]

**Pull request merged**